src/main/java/com/aca/exam/exam3/service/impl/users/plate/number/UserPlateNumberImpl.java

```
@@ -0,0 +1,80 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserPlateNumberImpl.class);
+
+    private final UserPlateNumberRepository userPlateNumberRepository;
+    private final UserService userService;
+    private final NumberPlateService numberPlateService;
+
+    public UserPlateNumberImpl(
+            UserPlateNumberRepository userPlateNumberRepository,
+            UserService userService,
+            NumberPlateService numberPlateService) {
+
+        this.userPlateNumberRepository = userPlateNumberRepository;
+        this.userService = userService;
+        this.numberPlateService = numberPlateService;
+    }
+
+    @Override
+    public UserPlateNumber create(CreateUserPlateNumberParams params) {
+        Assert.notNull(params, "User plate number creation params should not be null");
+        LOGGER.info("Creating user plate number with params - {}", params);
+
+        final User user = userService.getById(params.getUserId());
+
+        final NumberPlate numberPlate = numberPlateService.getById(params.getNumberPlateId());
+
+        final UserPlateNumber userPlateNumber = userPlateNumberRepository.save(
+                new UserPlateNumber(user, numberPlate)
+        );
+
+        LOGGER.info("Successfully created user plate number with params - {}, result - {}", params, userPlateNumber);
+        return userPlateNumber;
+    }
+
+    @Override
+    public List<UserPlateNumber> getByUserId(Long id) {
+        Assert.notNull(id, "User id should not be null");
+        LOGGER.info("Getting User plate numbers for user with id - {}", id);
+
+        final List<UserPlateNumber> userPlateNumberList = userPlateNumberRepository.findByUser(userService.getById(id));
+
+        LOGGER.info("Successfully got User plate numbers for user with id - {}, result - {}", id, userPlateNumberList);
+        return userPlateNumberList;
+    }
+
+    @Override
+    public UserPlateNumber getByNumberPlateId(Long id) {
```

the getByNumberPlateId should throw an exception if the userplatenumber not found. you can rename the method to findByNumberPlateId and return an optional